sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -2074,24 +2073,24 @@ test_suite(
     ],
 )
 
-# list of keys that will be used to build various flash images
-# it must contain at least one key of each type and contains all
+# List of keys that will be used to build various flash images.
+# It must contain at least one key of each type and contains all
 # the keys used in SIGVERIFY_LCS_2_VALID_KEY
 SIGVERIFY_LC_KEYS = [
-    FAKE_TEST_KEYS[0],
-    FAKE_DEV_KEYS[0],
-    FAKE_PROD_KEYS[0],
+    ALL_KEYS[KEY_ROLE.TEST][0],
+    ALL_KEYS[KEY_ROLE.DEV][0],
+    ALL_KEYS[KEY_ROLE.PROD][0],
 ]
 
-# provide a valid key for each LC state
-# this key must be present in SIGVERIFY_LC_KEYS to ensure
-# that the proper flush images are built
+# Provide a valid key for each LC state.
+# This key must be present in SIGVERIFY_LC_KEYS to ensure
+# that the proper flash images are built
 SIGVERIFY_LCS_2_VALID_KEY = {
-    "test_unlocked0": FAKE_TEST_KEYS[0],
-    "dev": FAKE_DEV_KEYS[0],
-    "prod": FAKE_PROD_KEYS[0],
-    "prod_end": FAKE_PROD_KEYS[0],
-    "rma": FAKE_PROD_KEYS[0],
+    "test_unlocked0": ALL_KEYS[KEY_ROLE.TEST][0],
+    "dev": ALL_KEYS[KEY_ROLE.DEV][0],
+    "prod": ALL_KEYS[KEY_ROLE.PROD][0],
+    "prod_end": ALL_KEYS[KEY_ROLE.PROD][0],
+    "rma": ALL_KEYS[KEY_ROLE.PROD][0],
```

Comment:
FYI, the TEST key is _also_ valid in the RMA state.

Reply:
I know that a TEST key is also valid in RMA: this is taken into account in  `KEY_ROLES_FOR_LC` in `rules/rom_keys.bzl`.
The purpose of the `SIGVERIFY_LCS_2_VALID_KEY` array is to implement `rom_e2e_sigverify_always` as documented in the test plan:
```
            - Prepare a ROM_EXT image with a valid signature generated using an authorized key.
            - Flip a bit/make all zeros/make all ones at random.
              - Log so that test can be reproduced.
            - Apply bootstrap pin strapping and reset the chip.
            - Load the image.
            - Verify that boot fails with the expected `BFV`: `kErrorSigverifyBadRsaSignature`.
            - Repeat the steps above for TEST, DEV, PROD, PROD_END, and RMA.
```
The testplan and test only checks **one** valid key per LC, and this PR does not change this behaviour.
Note that the `rom_e2e_sigverify_key_type` test already verifies all possible combinations of key type with LC state so it would be redundant to do it here as well. The purpose of this test is only to make sure that randomly flipping bits in the key makes the boot fail.